### PR TITLE
[Infineon] Provision to enable/disable NOC key-pair generation using HSM

### PR DESCRIPTION
**Pull request by ankk-css**
* Added the flag ENABLE_TRUSTM_NOC_KEYGEN in the config.
* Updated the functions to support NOC keygen with and without Trust M MTR.
* Restyled changes

### CODE REVIEW

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp [hunk 1]
```diff
@@ -136,11 +123,20 @@ CHIP_ERROR P256Keypair::Initialize(ECPKeyTarget key_target)
     }
     else
     {
+#if !ENABLE_TRUSTM_NOC_KEYGEN
+        if (CHIP_NO_ERROR == Initialize_H(this, &mPublicKey, &mKeypair))
+        {
+            mInitialized = true;
+        }
+        error = CHIP_NO_ERROR;
```
Comment (ksperling-apple) [inline src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp:131]: Why does this ignore errors from `Initialize_H` ?

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp [hunk 2]
```diff
@@ -188,20 +183,31 @@ CHIP_ERROR P256Keypair::ECDSA_sign_msg(const uint8_t * msg, size_t msg_length, P
...
+
+        // Api call to calculate the signature
+        return_status = trustm_ecdsa_sign(OPTIGA_KEY_ID_E0F0, digest, digest_length, signature_trustm, &signature_trustm_len);
     }
     else
     {
-        return_status = trustm_ecdsa_sign(OPTIGA_KEY_ID_E0F0, digest, digest_length, signature_trustm, &signature_trustm_len);
+#if !ENABLE_TRUSTM_NOC_KEYGEN
+        // Use the mbedtls based method
+        printf("ECDSA sing msg mbedtls\n");
```
Comment (ksperling-apple) [inline src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp:202]: This is probably meant to be `ChipLog`?

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp [hunk 3]
```diff
@@ -136,11 +123,20 @@ CHIP_ERROR P256Keypair::Initialize(ECPKeyTarget key_target)
     }
     else
     {
+#if !ENABLE_TRUSTM_NOC_KEYGEN
```
Comment (ksperling-apple) [inline src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp:126]: If you're having both if and else branches anyway, having the test not be negated would make the code easier to read.

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp [hunk 4]
```diff
@@ -87,36 +86,24 @@ static CHIP_ERROR get_trustm_keyid_from_keypair(const P256KeypairContext mKeypai
...
 {
-    // Add method to get the keyid
     if (CHIP_NO_ERROR != get_trustm_keyid_from_keypair(mKeypair, &keyid))
     {
         Clear();
     }
     else
     {
-        // Delete the key in SE
     }
```
Comment (ksperling-apple) [inline src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp:100]: Empty else branch? Also a comment why a key with a HSM key id doesn't need clearing would seem to make sense.

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp (hunk 1 above)
Comment (ankk-css) [reply src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp]: Updated this part.

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp (hunk 3 above)
Comment (ankk-css) [reply src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp:123]: Will make it more readable in the next PR. Thanks for the input.

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp (hunk 2 above)
Comment (ankk-css) [reply src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp]: Updated the code.

File: src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp (hunk 4 above)
Comment (ankk-css) [reply src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp]: We will look into this part . Thanks!

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**